packages/core/src/mesh/ModelMesh.ts

```diff
@@ -14,4 +58,783 @@ export class ModelMesh extends Mesh {
+      offset += 8;
+      elementCount += 2;
+    }
+    if (_uv5) {
+      vertexElements.push(new VertexElement("TEXCOORD_5", offset, VertexElementFormat.Vector2, 0));
+      offset += 8;
+      elementCount += 2;
+    }
+    if (_uv6) {
+      vertexElements.push(new VertexElement("TEXCOORD_6", offset, VertexElementFormat.Vector2, 0));
+      offset += 8;
+      elementCount += 2;
+    }
+    if (_uv7) {
+      vertexElements.push(new VertexElement("TEXCOORD_7", offset, VertexElementFormat.Vector2, 0));
+      offset += 8;
+      elementCount += 2;
+    }
+
+    this._elementCount = elementCount;
+    return vertexElements;
+  }
+
+  private _resetVertexArrayData(vertices: Float32Array): void {
+    // prettier-ignore
+    const { _elementCount,_vertexCount, _positions, _normals, _colors, _vertexChangeFlag, _weights, _joints, _tangents, _uv, _uv1, _uv2, _uv3, _uv4, _uv5, _uv6, _uv7 } = this;
+
+    if (_vertexChangeFlag & ValueChanged.Position) {
+      for (let i = 0; i < _vertexCount; i++) {
+        const start = _elementCount * i;
+        const position = _positions[i];
+        vertices[start] = position.x;
+        vertices[start + 1] = position.y;
+        vertices[start + 2] = position.z;
+      }
+    }
+
+    let offset = 3;
+
+    if (_normals) {
+      if (_vertexChangeFlag & ValueChanged.Normal) {
+        for (let i = 0; i < _vertexCount; i++) {
+          const start = _elementCount * i + offset;
+          const normal = _normals[i];
+          if (normal) {
+            vertices[start] = normal.x;
+            vertices[start + 1] = normal.y;
+            vertices[start + 2] = normal.z;
+          }
+        }
+      }
+      offset += 3;
+    }
+
+    if (_colors) {
+      if (_vertexChangeFlag & ValueChanged.Color) {
+        for (let i = 0; i < _vertexCount; i++) {
+          const start = _elementCount * i + offset;
+          const color = _colors[i];
+          if (color) {
+            vertices[start] = color.r;
+            vertices[start + 1] = color.g;
+            vertices[start + 2] = color.b;
+            vertices[start + 3] = color.a;
+          }
+        }
+      }
+      offset += 4;
+    }
+
+    if (_weights) {
+      if (_vertexChangeFlag & ValueChanged.Weight) {
+        for (let i = 0; i < _vertexCount; i++) {
+          const start = _elementCount * i + offset;
+          const weight = _weights[i];
+          if (weight) {
+            vertices[start] = weight.x;
+            vertices[start + 1] = weight.y;
+            vertices[start + 2] = weight.z;
+            vertices[start + 3] = weight.w;
+          }
+        }
+      }
+      offset += 4;
+    }
+
+    if (_joints) {
+      if (_vertexChangeFlag & ValueChanged.Joint) {
+        for (let i = 0; i < _vertexCount; i++) {
+          const start = _elementCount * i + offset;
+          const joint = _joints[i];
+          if (joint) {
+            vertices[start] = joint.x;
```

Comment:
joints is just bone index,use byte is enough. The internal big CPU and GPU Buffer can consider to use byte to save memory.